Telcodocs 1511 Enabling ALLMULTI flag (#68346)
openshift/openshift-docs: modules/nw-sriov-about-all-multi-cast-mode.adoc

// Module included in the following assemblies:
//
// * networking/hardware_networks/configuring-interface-sysctl-sriov-device.adoc

:_mod-docs-content-type: CONCEPT
[id="nw-about-all-multi-cast-mode_{context}"]
= About all-multicast mode

> Enabling all-multicast mode, particularly in the context of rootless applications, is important as without it, you would be required to grant the `NET_ADMIN` capability to the pod's Security Context Constraints (SCC).

Reviewer: Why is it preferred to enable all-multicast mode over granting `NET_ADMIN` capability to the pod's SSecurity Context Constraints?

Author: This is primarily associated with security concerns enabling NET_ADMIN capability provides extensive control over networking, including multi-cast, which might not be necessary for the pod's intended functionality. Allowing NET_ADMIN capability grants the pod privileges to make changes that extend beyond its specific requirements, potentially exposing security vulnerabilities.

Author: I did add a clarification

> Enabling all-multicast mode, particularly in the context of rootless applications, is important as without it, you would be required to grant the `NET_ADMIN` capability to the pod's Security Context Constraints (SCC). If you were to allow the `NET_ADMIN` capability to grant the pod privileges to make changes that extend beyond its specific requirements, you could potentially expose security vulnerabilities.

Reviewer 2: ```suggestion
Enabling all-multicast mode, particularly in the context of rootless applications, is critical. If you do not enable this mode, you would be required to grant the `NET_ADMIN` capability to the pod's Security Context Constraints (SCC). If you were to allow the `NET_ADMIN` capability to grant the pod privileges to make changes that extend beyond its specific requirements, you could potentially expose security vulnerabilities.
```
- Consider editing to break up the first long sentence.

> The tuning CNI plugin supports changing several interface attributes including the all-multicast mode. Enabling this mode allows DPDK applications running on Virtual Functions (VFs) configured on a SR-IOV network device to receive multicast traffic from applications on other VFs, whether attached to the same or different physical functions.

Reviewer: Spell out DPDK?

> The tuning CNI plugin supports changing several interface attributes including the all-multicast mode. Enabling this mode allows Data Plane Development Kit (DPDK) applications running on Virtual Functions (VFs) configured on a SR-IOV network device to receive multicast traffic from applications on other VFs, whether attached to the same or different physical functions.

Reviewer 3: remove DPDK, just applications in general

// Module included in the following assemblies:
//
// * networking/hardware_networks/configuring-interface-sysctl-sriov-device.adoc

:_mod-docs-content-type: CONCEPT
[id="nw-about-all-multi-cast-mode_{context}"]
= About allmulticast mode

Enabling allmulticast mode, particularly in the context of rootless applications, is important as without it, you would be required to grant the `NET_ADMIN` capability to the pod's Security Context Constraints (SCC).

> The tuning CNI plugin supports changing several interface attributes including the all-multicast mode. Enabling this mode allows rootless applications running on Virtual Functions (VFs) configured on a SR-IOV network device to receive multicast traffic from applications on other VFs.

Reviewer 4: Hi we also test between PFs and it works.

> The tuning CNI plugin supports changing several interface attributes including the all-multicast mode. Enabling this mode allows DPDK applications running on Virtual Functions (VFs) configured on a SR-IOV network device to receive multicast traffic from applications on other VFs, whether attached to the same or different physical functions.

Author: @gkopels I thought one of the test cases addressed VFs attached to different physical functions.

> The tuning CNI plugin supports changing several interface attributes including the all-multicast mode. Enabling this mode allows DPDK applications running on Virtual Functions (VFs) configured on a SR-IOV network device to receive multicast traffic from applications on other VFs, whether attached to the same or different physical functions.

Author: @mlguerrero12 Can I reinstate "Enabling this mode allows DPDK applications running on Virtual Functions (VFs) configured on a SR-IOV network device to receive multicast traffic from applications on other VFs, whether attached to the same or different physical functions." based on comment from @gkopels "Hi we also test between PFs and it works."

> The tuning CNI plugin supports changing several interface attributes including the all-multicast mode. Enabling this mode allows DPDK applications running on Virtual Functions (VFs) configured on a SR-IOV network device to receive multicast traffic from applications on other VFs, whether attached to the same or different physical functions.

Reviewer 3: if @gkopels verified this with intel and mellanox, then yes, it is good to put that back
